runtime/libc/wasi/include/wasi_backing.h

```diff
@@ -0,0 +1,1320 @@
+    __wasi_errno_t rc = __wasi_poll_oneoff(CURRENT_WASI_CONTEXT, in, out, nsubscriptions, &nevents);
+    if (rc != __WASI_ERRNO_SUCCESS)
+        goto done;
+
+    /* Write results to linear memory */
+    wasi_serdes_write_size_t(CURRENT_MEMORY_BASE, nevents_retptr, nevents);
+
+    for (__wasi_size_t i = 0; i < nsubscriptions; i++) {
+        wasi_serdes_write_event_t(CURRENT_MEMORY_BASE, out_baseptr, &out[i]);
+        out_baseptr += WASI_SERDES_SIZE_event_t;
+    }
+
+done:
+    return (uint32_t)rc;
+}
+
+/**
+ * Terminate the process normally. An exit code of 0 indicates successful
+ * termination of the program. The meanings of other values is dependent on
+ * the environment.
+ *
+ * @param exitcode
+ */
+__attribute__((noreturn)) void wasi_snapshot_preview1_proc_exit(__wasi_exitcode_t exitcode) {
+#ifdef LOG_WASI
+    fprintf(stderr, "wasi_snapshot_preview1_proc_exit(%u)\n", exitcode);
+#endif
+
+    __wasi_proc_exit(CURRENT_WASI_CONTEXT, exitcode);
+    assert(0);
+}
+
+/**
+ * Send a signal to the process of the calling thread.
+ *
+ * @param sig The signal condition to trigger.
+ * @return status code
+ */
+uint32_t wasi_snapshot_preview1_proc_raise(uint32_t sig) {
+#ifdef LOG_WASI
+    fprintf(stderr, "wasi_snapshot_preview1_proc_raise(%u)\n", sig);
+#endif
+
+    __wasi_errno_t rc;
+
+    if (sig > UINT8_MAX)
+        goto err_sig_overflow;
+
+    rc = __wasi_proc_raise(CURRENT_WASI_CONTEXT, (__wasi_signal_t)sig);
+
+done:
+    return (uint32_t)rc;
+err_sig_overflow:
+    rc = __WASI_ERRNO_INVAL;
+    goto done;
+}
+
+/**
+ * Write high-quality random data into a buffer.
+ * This function blocks when the implementation is unable to immediately
+ * provide sufficient high-quality random data.
+ * This function may execute slowly, so when large mounts of random data are
+ * required, it's advisable to use this function to seed a pseudo-random
+ * number generator, rather than to provide the random data directly.
+ *
+ * @param buf_baseretptr The buffer to fill with random data.
+ * @param buf_len The length of the buffer
+ * @return status code
+ */
+uint32_t wasi_snapshot_preview1_random_get(__wasi_size_t buf_baseretptr, __wasi_size_t buf_len) {
+#ifdef LOG_WASI
+    fprintf(stderr, "wasi_snapshot_preview1_random_get(%u,%u)\n", buf_baseretptr, buf_len);
+#endif
+
+    /* Check Bounds */
+    wasi_serdes_check_bounds(buf_baseretptr, CURRENT_MEMORY_SIZE, buf_len);
+
+    /* Write random bytes directly to linear memory */
+    return (uint32_t)__wasi_random_get(CURRENT_WASI_CONTEXT, &CURRENT_MEMORY_BASE[buf_baseretptr], buf_len);
+}
+
+/**
+ * Temporarily yield execution of the calling thread similar to `sched_yield` in POSIX.
+ * This implementation ignores client calls and silently returns RC 0
+ *
+ * @return WASI_ESUCCESS
+ */
+uint32_t wasi_snapshot_preview1_sched_yield(void) {
+#ifdef LOG_WASI
+    fprintf(stderr, "wasi_snapshot_preview1_sched_yield()\n");
+#endif
+
+    return (uint32_t)__wasi_sched_yield(CURRENT_WASI_CONTEXT);
```

Comment:
Is all of awsm using 4 space indentation?  I thought it was our normal 8. If it all was unified around 4, so be it.

Reply:
It's been four spaces. Here's a link to the repo before I touched it: https://github.com/gwsystems/aWsm/blob/2d654da6d52a95ec140845c7eb02943db67dfbee/runtime/runtime.c. I believe that SLEdge continued this pattern as well. There are stylistic differences in here compared to SLEdge compared to Composite. My priorities with all this have mostly been just focused on getting clang-format hooked up with an auto-format script and a CI check, and as much as possible, I've tried to match the existing de-facto standards in place to minimize resistance to getting the tooling in place.

Reply:
It would be relatively simple to tweak the clang-format setup to make this standardized in the future if you wish.

Reply:
Rust standard is 4 spaces, so I matched that. C code bases are usually 4 or 8. Composite I think is actually tab intended. My preference is to keep this space indented at 4. The code style is intentionally "rusty" and not supposed to be the same as Composite (that takes a lot of ques from Linux)